src/metrics_server/model.ts

```diff
@@ -0,0 +1,18 @@
+// Copyright 2020 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Generic table interface that supports row insertion.
+export interface InsertableTable<T> {
+  insert(rows: T|T[]): Promise<void>;
```

Comment:
This type of type overloading makes things harder to reason about. Let's stick to an array, or provide two methods.

Reply:
Defined a single method that accepts an array.